src/githubClient.ts

```diff
@@ -0,0 +1,30 @@
+import * as core from '@actions/core';
+import { WebRequest, WebResponse, sendRequest } from "./utilities/httpClient";
+
+export class GitHubClient {
+    constructor(repository: string, token: string) {
+        this._repository = repository;
+        this._token = token;
+    }
+
+    public async getWorkflows(force?: boolean): Promise<any> {
+        if (force || !this._workflowsPromise) {
```

Comment:
I don't see a need for caching given that it gets called from a single place.

Reply:
Added it, keeping in mind if we need to call API from multiple places. Want me to remove it?

Reply:
Remove it, if this flow is not being hit. :)